core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -152,8 +152,8 @@ class LogCleaner(val config: CleanerConfig,
    * For testing, a way to know when work has completed. This method blocks until the
    * cleaner has processed up to the given offset on the specified topic/partition
    */
-  def awaitCleaned(topic: String, part: Int, offset: Long, timeout: Long = 30000L): Unit = {
-    while(!cleanerManager.allCleanerCheckpoints.contains(TopicAndPartition(topic, part)))
+  def awaitCleaned(topic: String, part: Int, offset: Long): Unit = {
```

Comment:
Chiming late here: I also feel setting an infinite timeout value is generally not a good idea since IF there are any issue causing the test to block, it is hard to detect this issue: for example today we also encounter blocking-forever test cases in Jenkins from time to time, and because we set the Jenkins time to 1 hour (? not sure if the value is exact), we ended up seeing an "timed out" failed Jenkins without much information which test case caused it.

I think the current approach of setting a large enough default value is a better option.